tools/functional-tester/etcd-tester/lease_stresser.go

```diff
@@ -0,0 +1,377 @@
+	}
+
+	plog.Printf("lease: %v is not expired", leaseID)
+	return false, nil
+}
+
+func (ls *leaseStresser) hasKeysAttachedToLeaseExpired(ctx context.Context, leaseID int64) (bool, error) {
+	// The keys attached to the lease has the format of "<leaseID>_<idx>" where idx is the ordering key creation
+	// Since the format of keys contains about leaseID, finding keys base on "<leaseID>" prefix
+	// determines whether the attached keys for a given leaseID has been deleted or not
+	resp, err := ls.kvc.Range(ctx, &pb.RangeRequest{
+		Key:      []byte(fmt.Sprintf("%d", leaseID)),
+		RangeEnd: []byte(clientv3.GetPrefixRangeEnd(fmt.Sprintf("%d", leaseID))),
+	}, grpc.FailFast(false))
+
+	if err != nil {
+		plog.Printf("get Key info ERROR %v \n", err)
+		return false, err
+	}
+	return len(resp.Kvs) == 0, nil
+}
+
+func (ls *leaseStresser) createLeaseAndKeepAlive(ctx context.Context) (int64, error) {
+	leaseID, err := ls.createLease(ctx)
+	if err != nil {
+		return -1, err
+	}
+	go ls.keepLeaseAlive(ctx, leaseID)
+	return leaseID, nil
+}
+
+func (ls *leaseStresser) createLease(ctx context.Context) (int64, error) {
+	resp, err := ls.lc.LeaseGrant(ctx, &pb.LeaseGrantRequest{
+		TTL: TTL,
+	})
+
+	if err != nil {
+		return -1, err
+	}
+
+	return resp.ID, nil
+}
+
+func (ls *leaseStresser) keepLeaseAlive(ctx context.Context, leaseID int64) {
+
+	for {
+		select {
+		case <-time.After(1 * time.Second):
+		case <-ctx.Done():
+			plog.Println("Stresser %v keepalive context closed")
+			return
+		}
+		stream, err := ls.lc.LeaseKeepAlive(ctx, grpc.FailFast(false))
+		if err != nil {
+			plog.Printf("keepLeaseAlive ERROR %v", err)
+			return
+		}
+
+		err = stream.Send(&pb.LeaseKeepAliveRequest{ID: leaseID})
+		if err != nil {
+			plog.Printf("keepLeaseAlive stream send lease %v ERROR %v ", leaseID, err)
+			continue
+		}
+
+		respRC, err := stream.Recv()
+
+		if err != nil {
+			plog.Printf("keepLeaseAlive stream recieve lease %v stream ERROR %v ", leaseID, err)
+			continue
+		}
+		// lease expires after TTL become 0
+		// don't send keepalive if the lease has expired
+		if respRC.TTL <= 0 {
+			plog.Printf("keepLeaseAlive: lease %v has expired ", respRC.ID)
+			return
+		}
+	}
+}
+
+// attachKeysWithLease function attaches keys to the lease.
+// the format of key is the concat of leaseID + '_' + '<order of key creation>'
+// e.g 5186835655248304152_0 for first created key and 5186835655248304152_1 for second created key
+func (ls *leaseStresser) attachKeysWithLease(ctx context.Context, leaseID int64) error {
+	for j := 0; j < ls.keysPerLease; j++ {
+		_, err := ls.kvc.Put(ctx, &pb.PutRequest{
+			Key:   []byte(fmt.Sprintf("%d%s%d", leaseID, "_", j)),
+			Value: []byte(fmt.Sprintf("bar")),
+			Lease: leaseID,
+		}, grpc.FailFast(false))
+
+		if err != nil {
+			plog.Printf("attach keys to lease %v ERROR %v", leaseID, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func (ls *leaseStresser) randomlyDropLease(ctx context.Context, leaseID int64) bool {
```

Comment:
maybeDropLease(ctx context, leaseID)?

also move the code comment to the method comment?